tests/core/fsck-on-boot/task.yaml

```diff
@@ -1,24 +1,37 @@
 summary: the boot base provides essential fsck programs
+
 details: |
   Snapd uses vfat on certain essential boot partitions, due to external
   requirements imposed by the bootloader architecture. This test verifies that
   the boot process is capable of detecting unclean vfat and fixing it before
   such file system is mounted. This is an essential property to ensure
   longevity of devices that rely on write to vfat to operate.
+
 prepare: |
   tests.cleanup prepare
+
 execute: |
   unmount_vfat() {
     if os.query is-core16; then
-      # Refer to the core 16 PC gadget for details:
+      # Refer to the core 16 gadgets for details:
       # https://github.com/snapcore/pc-amd64-gadget/blob/16/gadget.yaml
-      umount /boot/efi
-      umount /boot/grub
+      # https://github.com/snapcore/pi2-gadget/blob/master/gadget.yaml
+      if [[ "$SPREAD_SYSTEM" == ubuntu-core-16-arm-* ]]; then
+        umount /boot/uboot
+      else
+        umount /boot/efi
+        umount /boot/grub
+      fi
     elif os.query is-core18; then
-      # Refer to the core 18 PC gadget for details:
+      # Refer to the core 18 gadgets for details:
       # https://github.com/snapcore/pc-amd64-gadget/blob/18/gadget.yaml
-      umount /boot/efi
-      umount /boot/grub
+      # https://github.com/snapcore/pi2-gadget/blob/18/gadget.yaml
+      if [[ "$SPREAD_SYSTEM" == ubuntu-core-18-arm-* ]]; then
+        umount /boot/uboot
+      else
+        umount /boot/efi
+        umount /boot/grub
+      fi
     elif os.query is-core20; then
       # TODO:UC20 The property of having to keep a mounted vfat at all time
       # is not the most fortunate. Any power loss will result in a dirty
```

Comment:
maybe add a link to the uc20 gadgets here too? see https://github.com/snapcore/pi-gadget/blob/20-arm64/gadget.yaml and https://github.com/snapcore/pc-amd64-gadget/blob/20/gadget.yaml

Reply:
done